grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -160,7 +165,12 @@ class StockMovementController {
 
     def removeStockMovement = {
         StockMovement stockMovement = stockMovementService.getStockMovement(params.id)
-        if (stockMovement?.shipment?.currentStatus == ShipmentStatusCode.PENDING || !stockMovement?.shipment?.currentStatus) {
+        if (stockMovement?.requisition?.origin?.id != session.warehouse.id) {
+            if (params.show) {
+                redirect(action: "show", id: params.id)
```

Comment:
This needs to be followed by a return otherwise this could fall through to the redirect to the list action at the bottom, which might still redirect to the right place but will likely cause an error like "the response has already been committed".